pex/platforms.py

```diff
@@ -16,8 +21,25 @@ class InvalidPlatformError(Exception):
 
     SEP = "-"
 
+    __slots__ = ("platform", "impl", "version", "abi")
+
+    def __init__(self, platform, impl, version, abi):
+        super(Platform, self).__init__()
+        if not all((platform, impl, version, abi)):
```

Comment:
It'd be better to check if these are `str`, rather than simply being truthy. But I tried to minimize my changes. Lmk if I should add it.

Reply:
I think we might want to make `abi` at least into an enum when we can drop py2, although I'm not sure checking for `str` now makes it harder to do later. I think leaving it as you've got it here sounds good.